src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -220,22 +222,45 @@ public void setRatedIfNotExceeded(@Nullable ZonedDateTime exerciseDueDate, Zoned
      * - no due date is set OR
      * - the submission type is INSTRUCTOR / TEST
      *
-     * @param exerciseDueDate date after which no normal submission is considered rated.
-     * @param submission      to which the result belongs.
-     * @param participation   to wich the submission belongs
+     * @param dueDate    due date of the exercise or individual due date if applicable
+     * @param submission to which the result belongs.
      */
-    public void setRatedIfNotExceeded(ZonedDateTime exerciseDueDate, Submission submission, Participation participation) {
+    public void setRatedIfNotAfterDueDate(@Nullable ZonedDateTime dueDate, @NotNull Submission submission) {
         if (submission.getType() == SubmissionType.INSTRUCTOR || submission.getType() == SubmissionType.TEST) {
             this.rated = true;
         }
         else if (submission.getType() == SubmissionType.ILLEGAL || participation.isTestRun()) {
```

Comment:
Why do you use the `participation` from the current `Result` object, but not the `submission`? 🤔 They are both attributes of `Result`, but you have only removed one of the parameters.

Reply:
If I'm not mistaken, this method is always effectively called with `this.submission` and `this.participation`. Can these parameters both be removed?

Reply:
My goal was to move the `ExerciseDateService.getDueDate` call into the methods to make sure that always the individual due date gets used. I'll double-check if all these methods are needed.